test/fixtures/bem-nesting.css

```diff
@@ -0,0 +1,20 @@
+/** @define block-name */
+
+.block-name {
+  color: green;
+
+  .block-name__element {}
```

Comment:
Do you mean it should fail the actual linter? Currently this passes fine:

``` css
.Component .Component-elem {
  color: red;
}
```

Whilst it maybe a bad practice, there are no specific rules in BEM or SUIT to say you can't prefix selectors with the component name.